Libraries/Components/CheckBox/CheckBox.android.js

```diff
@@ -80,85 +135,65 @@ type DefaultProps = {
  * @keyword checkbox
  * @keyword toggle
  */
-let CheckBox = createReactClass({
-  displayName: 'CheckBox',
-  propTypes: {
-    ...DeprecatedViewPropTypes,
-    /**
-     * The value of the checkbox.  If true the checkbox will be turned on.
-     * Default value is false.
-     */
-    value: PropTypes.bool,
-    /**
-     * If true the user won't be able to toggle the checkbox.
-     * Default value is false.
-     */
-    disabled: PropTypes.bool,
-    /**
-     * Used in case the props change removes the component.
-     */
-    onChange: PropTypes.func,
-    /**
-     * Invoked with the new value when the value changes.
-     */
-    onValueChange: PropTypes.func,
-    /**
-     * Used to locate this view in end-to-end tests.
-     */
-    testID: PropTypes.string,
-  },
-
-  getDefaultProps: function(): DefaultProps {
-    return {
-      value: false,
-      disabled: false,
-    };
-  },
+class CheckBox extends React.Component<Props> {
+  _nativeRef: ?React.ElementRef<CheckBoxNativeType> = null;
+  _setNativeRef = setAndForwardRef({
```

Comment:
We should flow type `setAndForwardRef` to ensure that `getForwardedRef` and `setLocalRef` operate on the same ref type.

Reply:
Separate is best. I just noticed as part of some weirdness in how it is being used here.